states/HI/governor.yaml

```
@@ -0,0 +1,317 @@
+            "Turkey": "Turkey"
+            "Turkmenistan": "Turkmenistan"
+            "Tuvalu": "Tuvalu"
+            "Uganda": "Uganda"
+            "Ukraine": "Ukraine"
+            "United Arab Emirates": "United Arab Emirates"
+            "United Kingdom": "United Kingdom"
+            "United States": "United States"
+            "Uruguay": "Uruguay"
+            "Uzbekistan": "Uzbekistan"
+            "Vanuatu": "Vanuatu"
+            "Vatican City": "Vatican City"
+            "Venezuela": "Venezuela"
+            "Vietnam": "Vietnam"
+            "Virgin Islands, British": "Virgin Islands, British"
+            "Virgin Islands, U.S.": "Virgin Islands, U.S."
+            "Yemen": "Yemen"
+            "Zambia": "Zambia"
+            "Zimbabwe": "Zimbabwe"
+        - name: "input_10"
+          selector: "#input_1_10"
+          value: "Other"
+          required: true
+          options:
+            "Comment/ Suggestion": "Comment/ Suggestion"
+            "Complaint": "Complaint"
+            "Question": "Question"
+            "Other": "Other"
+    - fill_in:
+        - name: "input_11"
+          selector: "#input_1_11"
+          value: $MESSAGE
+          required: true
+    # The address county is added temporarily so that the JS that follows can grab it and use it.  It is then cleared once the JS is done with it.
+    - fill_in:
+        - name: "input_9.2"
+          selector: "#input_1_9_2"
+          value: $ADDRESS_COUNTY
+    - javascript:
+        # Hawaii has a 'region' dropdown that doesn't correlate directly to any address component we currently collect.  So we have connected the region to a county where we can, and a zip code where we cannot.  The JS below handles this and selects the appropriate option value.
+        # NOTE: The apostrophes in the values of the countyRegionXref hash (the region names) are right apostrophes (Unicode U+2019).  The regular straight apostrophe (Unicode U+0027) will not match on the form.
+        - value: "var county = document.getElementById('input_1_9_2').value;
+        var zip = document.getElementById('input_1_9_5').value;
+
+        const zipList = new Set(['96719', '96755', '96743', '96738', '96740', '96725', '96750', '96704', '96722', '96726', '96763']);
+
+        const countyRegionXref = {
+          'Hawaii' : 'East Hawaiʻi Island',
+          'Honolulu' : 'Oʻahu',
+          'Kalawao' : 'Molokaʻi',
+          'Kauai' : 'Kauaʻi',
+          'Maui' : 'Maui'
+        }
+
+        function findRegion(county, zip) {
+          var region;
+          if (zip == '96763') {
+            var region = 'Lanaʻi';
+          }
+          else if (zipList.has(zip)) {
+            var region = 'West Hawaiʻi Island';
+          }
+          else {
+            var region = countyRegionXref[county];
+          }
+          return region;
+        }
+
+        document.getElementById('input_1_18').value = findRegion(county, zip);
+        document.getElementById('input_1_9_2').value ='';"
```

Would this work here?

```suggestion
        county.value ='';"
```

---

That doesn't work because ```county``` is just a string containing the value of the field.
```var county = document.getElementById('input_1_9_2').value;```
I could change that so that the var that now represents county instead be the element, then pass the value attribute to the function, and then the above suggestion would work.  Not sure which way would be better?

---

Ah, right.  I don't have a strong preference, though I think it's more clear to assign the element once and operate on it than grabbing it multiple times.  Makes future changes a bit more error-prone.

---

Ok, let me change it up.